src/iop/cacorrectrgb.c

```diff
@@ -399,23 +399,35 @@ dt_omp_firstprivate(in, blurred_in, manifold_lower, manifold_higher, width, heig
         {
           const size_t c = (guide + kc + 1) % 3;
           const float pixel = fmaxf(in[k * 4 + c], 1E-6f);
-          const float log_diff = logf(pixel) - pixelg;
+          const float log_diff = log2f(pixel) - pixelg;
           manifold_higher[k * 4 + c] = log_diff * w;
         }
         manifold_higher[k * 4 + guide] = fmaxf(in[k * 4 + guide], 0.0f) * w;
         manifold_higher[k * 4 + 3] = w;
+        // manifold_lower still contains the values from first iteration
+        // -> reset it.
+        for(size_t c = 0; c < 4; c++)
```

Comment:
you should be able to use: for_four_channels()

it is a macro that will vectorize this loop if possible.

Reply:
EDIT: not possible as the loop is already inside a omp/simd section.